src/librustdoc/clean/types.rs

```diff
@@ -2002,8 +2001,7 @@ impl Path {
     }
 
     crate fn whole_name(&self) -> String {
-        String::from(if self.global { "::" } else { "" })
-            + &self.segments.iter().map(|s| s.name.to_string()).collect::<Vec<_>>().join("::")
+        self.segments.iter().map(|s| s.name.to_string()).collect::<Vec<_>>().join("::")
```

Comment:
Fixed, thanks!

Reply:
Actually, I think `intersperse` might be *less* efficient in this case because it will be cloning the `::`, whereas `join` seems to not clone the separator. Perhaps we should just leave it as-is?

Reply:
:shrug: I think intersperse is easier to read, but I don't feel strongly, whatever you have currently is fine.